src/main/scala/zio/prelude/IdentityBoth.scala

```diff
@@ -65,6 +68,43 @@ object IdentityBoth extends LawfulF.Invariant[DeriveEqualIdentityBothInvariant,
   def apply[F[_]](implicit identityBoth: IdentityBoth[F]): IdentityBoth[F] =
     identityBoth
 
+  /**
+   * The `IdentityBoth` instance for `Either`.
+   */
+  implicit def EitherIdentityBoth[L]: IdentityBoth[({ type lambda[+r] = Either[L, r] })#lambda] =
+    new IdentityBoth[({ type lambda[+r] = Either[L, r] })#lambda] {
+      val any: Either[L, Any] = Right(())
+
+      def both[A, B](fa: => Either[L, A], fb: => Either[L, B]): Either[L, (A, B)] =
+        fa.flatMap(a => fb.map(b => (a, b)))
+    }
+
+  /**
+   * The `IdentityBoth` instance for a failed `Either`
+   */
+  implicit def EitherFailedIdentityBoth[R]: IdentityBoth[({ type lambda[+l] = Failure[Either[l, R]] })#lambda] =
+    new IdentityBoth[({ type lambda[+l] = Failure[Either[l, R]] })#lambda] {
+      val any: Failure[Either[Any, R]] = Failure.wrap(Left(()))
+
+      def both[A, B](fa: => Failure[Either[A, R]], fb: => Failure[Either[B, R]]): Failure[Either[(A, B), R]] =
+        Failure.wrap {
+          Failure
+            .unwrap(fa)
+            .left
+            .flatMap(a => Failure.unwrap(fb).left.map(b => (a, b)))
+        }
+    }
+
+  /**
+   * The `IdentityBoth` instance for `Future`.
+   */
+  implicit def FutureIdentityBoth: IdentityBoth[Future] =
```

Comment:
Can be a `val` here since we don't need an implicit `ExecutionContext` for `zip`.

Reply:
Ohhh yeah, right, thanks!